core/src/main/java/io/questdb/griffin/engine/functions/regex/RegexpReplaceVarcharFunctionFactory.java

```diff
@@ -93,8 +99,207 @@ public Function newInstance(
         return new Func(value, pattern, patternPos, replacement, replacementPos, maxLength, position);
     }
 
+    /**
+     * Returns true if the input pattern can operate on a UTF-16 view of a UTF-8 string,
+     * i.e. it only cares about ASCII chars.
+     */
+    private boolean canSkipUtf8Decoding(CharSequence pattern) {
+        return Chars.isAscii(pattern)
+                && !Chars.contains(pattern, "\\D") && !Chars.contains(pattern, "[^0-9]")
```

Comment:
What if the pattern contains `"^[0-8]`"? I think if the pattern contains `^` it may match non-ascii.

Reply:
Good catch.

> I think if the pattern contains `^` it may match non-ascii.

This may be too string. E.g. `[^/]+` is fine since it allows multiple chars. Let me think about how to improve the check.

Reply:
Tried to improve the check in 96d5e33fe93e3525995fd1ab175466386ab24901

Now, we give up on all character classes + quantifiers, but `[...]+`.

@jerrinot would be great if you could also take a look and tell me if you see flaws in the check.

Reply:
can we do the entire check in single pass over the pattern?

Reply:
also, this method could be made static and tested in isolation on a bunch of patters, perhaps we can pinch patters from the interwebs ?

Reply:
Good ideas. Let me apply them.

Reply:
Changed to single pass and added a test in e376abceda582bedfd56f0d68dce2a5ce5713376

The test uses simple patterns that cover all assumed cases.